backend-api/template.yaml

```diff
@@ -225,4 +270,77 @@ Resources:
               - Effect: Allow
                 Action:
                   - kms:Sign
-                Resource: !ImportValue platform-kms-signing-key-arn
\ No newline at end of file
+                Resource: !ImportValue platform-kms-signing-key-arn
+
+# TxMA Audit service
+  TxMASQSQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 604800
+      VisibilityTimeout: 60
+      KmsMasterKeyId: !Ref TxMAKMSEncryptionKey
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt TxMASQSQueueDeadLetterQueue.Arn
+        maxReceiveCount: 5
+
+
+# TxMA account consumes this queue
+  TxMASQSQueuePolicy:
+    Type: AWS::SQS::QueuePolicy
+    Properties:
+      Queues:
+        - !Ref TxMASQSQueue
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Action:
+              - "sqs:DeleteMessage"
+              - "sqs:GetQueueAttributes"
+              - "sqs:ChangeMessageVisibility"
+              - "sqs:ReceiveMessage"
+            Resource:
+              - !GetAtt TxMASQSQueue.Arn
+            Principal:
+              AWS:
+                - !FindInMap [TxMA, !Ref Environment, TxmaAccount]
+
+  TxMASQSQueueDeadLetterQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      MessageRetentionPeriod: 259200 # three days
+      KmsMasterKeyId: !Sub TxMAKMSEncryptionKey
+
+  TxMAKMSEncryptionKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: A KMS Key for encrypting the SQS Queue for TxMA
+      Enabled: true
+      KeyPolicy:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action:
+              - "kms:*"
+            Resource:
+              - "*"
+          - Effect: Allow
+            Principal:
+              AWS: !FindInMap [TxMA, !Ref Environment, TxmaAccount]
+            Action:
+              - "kms:Encrypt"
+              - "kms:Decrypt"
+              - "kms:GenerateDataKey"
+            Resource:
+              - "*"
+      KeySpec: SYMMETRIC_DEFAULT
+      KeyUsage: ENCRYPT_DECRYPT
+      MultiRegion: false
+      PendingWindowInDays: 7
+      Tags:
```

Comment:
Is there a tagging strategy?

Reply:
I currently don't have one. I'm not sure if there's a programme wide one. It would be nice to look up some best practices and pick what works for us.

Reply:
https://gds-way.digital.cabinet-office.gov.uk/manuals/aws-tagging.html#tagging-aws-resources

Reply:
@dothomson -> I think the sam app inherits tags from the pipeline. the stack orchestration tool uses the` aws cloudformation create-stack` command, which takes in a `--tags` argument.

I _assume_ the resources in the sam app also inherit the tags if the resources are tag-able

Potentially something we should be doing in tf instead of the template.yaml

Reply:
I've taken tags out as I think these should be provided at a stack level and inherited from the pipelines